### PR TITLE
fix: minfee key table not registered on app restarts

### LINKED ISSUE #3553: GlobalMinGasPrice not getting persisted
### Summary of Bug

Apparently, there is a problem persisting the app version and the GlobalMinGasPrice. I will continue digging.

```
celestia-appd tx blob pay-for-blob 0x00010203040506070809 0x000102030405060708094314314324324234234234234234234324234234234232 --fees 21000utia --chain-id 
private --keyring-backend test -b block --yes --from validator
code: 111222
codespace: undefined
data: ""
events: []
gas_used: "8786"
gas_wanted: "0"
height: "0"
info: ""
logs: []
raw_log: "recovered: parameter GlobalMinGasPrice not registered\ncaused by transaction:\n*types.MsgPayForBlobs{signer:\"celestia147x5c0hxx4vt7j8njx4s000vwut8vlxm7mz5ua\"
  namespaces:\"\\000\\000\\000\\000\\000\\000\\000\\000\\000\\000\\000\\000\\000\\000\\000\\000\\000\\000\\000\\000\\001\\002\\003\\004\\005\\006\\007\\010\\t\"
  blob_sizes:33 share_commitments:\"w3\\223!\\357\\222\\346$\\206\\022dp\\244ARt\\373\\\\\\356@\\266\\210\\\"\\206\\363\\212\\2430\\r\\224\\320\\366\"
  share_versions:0 }\n\nstack:\ngoroutine 468 [running]:\nruntime/debug.Stack()\n\t/Users/midnight/go/go1.22.3/src/runtime/debug/stack.go:24
  +0x64\ngithub.com/cosmos/cosmos-sdk/baseapp.newDefaultRecoveryMiddleware.func1({0x107789c80,
  0x14001665e70})\n\t/Users/midnight/go2/pkg/mod/github.com/celestiaorg/cosmos-sdk@v1.22.0-sdk-v0.46.16/baseapp/recovery.go:71
  +0x2c\ngithub.com/cosmos/cosmos-sdk/baseapp.newRecoveryMiddleware.func1({0x107789c80,
  0x14001665e70})\n\t/Users/midnight/go2/pkg/mod/github.com/celestiaorg/cosmos-sdk@v1.22.0-sdk-v0.46.16/baseapp/recovery.go:39
  +0x4c\ngithub.com/cosmos/cosmos-sdk/baseapp.processRecovery({0x107789c80, 0x14001665e70},
  0x1400023c780)\n\t/Users/midnight/go2/pkg/mod/github.com/celestiaorg/cosmos-sdk@v1.22.0-sdk-v0.46.16/baseapp/recovery.go:28
  +0x50\ngithub.com/cosmos/cosmos-sdk/baseapp.processRecovery({0x107789c80, 0x14001665e70},
  0x1400140d8a8)\n\t/Users/midnight/go2/pkg/mod/github.com/celestiaorg/cosmos-sdk@v1.22.0-sdk-v0.46.16/baseapp/recovery.go:33
  +0xb8\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).runTx
```

### Version

2

### Steps to Reproduce

1. checkout v2.0.0-rc1
2. bash scripts/single-node.sh
3. let it run past block 3
4. stop celestia-appd
5. run it again using:
```
celestia-appd start \
  --api.enable \
  --grpc.enable \
  --grpc-web.enable \
  --v2-upgrade-height 3
```
6. execute:
```
celestia-appd tx blob pay-for-blob 0x00010203040506070809 0x000102030405060708094314314324324234234234234234234324234234234232 --fees 21000utia --chain-id 
private --keyring-backend test -b block --yes --from validator
```

Second one, if you do the above steps, and at step 5, remove the --v2-upgrade-height 3 flag, and execute the PFB command:

```
celestia-appd tx blob pay-for-blob 0x00010203040506070809 0x000102030405060708094314314324324234234234234234234324234234234232 --fees 21000utia --chain-id 
private --keyring-backend test -b block --yes --from validator
code: 37
codespace: sdk
data: ""
events: []
gas_used: "8786"
gas_wanted: "-1"
height: "0"
info: ""
logs: []
raw_log: 'app version 0 is not supported: feature not supported'
timestamp: ""
tx: null
txhash: C586BBD76F4AAFBFAD7064472A677AB8FC5D0E0429AFBBC29A05DB91550E2C35
```

### PR DESCRIPTION
**Pull request by ninabarbakadze**
## Overview

Fixes #3553

### CODE REVIEW

**Review by cmwaters — COMMENTED** [review]
It's common for modules to have the `RegisterMinFeeParamTable` or equivalent in the constructor of the keeper. Since `minfee` doesn't have a keeper per say (there is nothing in `minfee` that is committed to state), perhaps it's cleaner to include it in the constructor of the module itself. That way we know whenever the module is used the param should be registered rather than have it haphazardly chucked somewhere in `app.go`

https://github.com/celestiaorg/celestia-app/blob/34f2443ba639e68a489307a75dcab84d3030c40b/x/minfee/module.go#L79-L84

File: app/app_test.go [hunk 1]
```diff
@@ -39,6 +40,11 @@ func TestNew(t *testing.T) {
 	t.Run("should not have sealed the baseapp", func(t *testing.T) {
 		assert.False(t, got.IsSealed())
 	})
+	t.Run("should have set the minfee key table", func(t *testing.T) {
+		subspace := got.GetSubspace(minfee.ModuleName)
+		hasKeyTable := subspace.HasKeyTable()
+		assert.True(t, hasKeyTable)
+	})
```
Comment (rootulp) [inline app/app_test.go:47]: 👍

**Review by rootulp — COMMENTED** [review]
+1 to Callum's comment: I think it would be clearer if the module registered it's own param table. Then we can add a unit test to to the x/minfee module to ensure it always registered it's own param table (in addition) to the unit test you added to app_test.go

**Comment by ninabarbakadze** [comment]
> ## Overview
> Fixes #3553



> +1 to Callum's comment: I think it would be clearer if the module registered it's own param table. Then we can add a unit test to to the x/minfee module to ensure it always registered it's own param table (in addition) to the unit test you added to app_test.go

Addressed your recommendations 👍🏼

File: x/minfee/module_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	stateStore.MountStoreWithDB(tStoreKey, storetypes.StoreTypeTransient, nil)
+	require.NoError(t, stateStore.LoadLatestVersion())
+
+	registry := codectypes.NewInterfaceRegistry()
+
+	// Create a params keeper
+	paramsKeeper := paramkeeper.NewKeeper(codec.NewProtoCodec(registry), codec.NewLegacyAmino(), storeKey, tStoreKey)
+	subspace := paramsKeeper.Subspace(minfee.ModuleName)
+
+	// Initialize the minfee module which register set the key table
```
Comment (rootulp) [inline x/minfee/module_test.go]:
```suggestion
	// Initialize the minfee module which registers the key table
```

**Review by rootulp — APPROVED** [review]
Thanks for fixing!

File: x/minfee/module_test.go (hunk 2 above)
Comment (ninabarbakadze) [reply x/minfee/module_test.go]: it's very strange to me how sometimes i push a commit fixing/refactoring a piece of code that you still see later on. for example i fixed that comment here https://github.com/celestiaorg/celestia-app/pull/3571/commits/563355e919493e4923b8419b1eae4320f4aad064 and somehow it still showed you an old version 🤔

Comment (rootulp) [reply x/minfee/module_test.go]: oh hmm I have no idea. I've seen that happen if I start reviewing a PR while the author pushes a commit and I leave a review before refreshing to view what they pushed after I started reviewing.

But in this case it looks like you pushed that commit 3 hours ago and I reviewed ~24 mins ago so I should've seen the latest changes 🤷

**Comment by ninabarbakadze** [comment]
> Thanks nina!
> 
> We should probably look to turn the construction of the state store (needed to initialise the keeper) into some test helper as I see it used often

np! yeah, good idea. exporting into an issue.

**Pull request merged**